### PR TITLE
Handle falsey values

### PR DESCRIPTION
**Pull request by kahummer**
closes #414 
https://github.com/onaio/gisida/pull/414 has many conflicts to attend to I realized the only changes being added are on the popup the pr seeks get https://github.com/onaio/gisida/pull/414 code merged to staging

### CODE REVIEW

**Comment by cKellyDesign** [comment]
@kahummer could you point out the lines of the actual change here that's not just formatting changes?

File: src/map/addPopUp.js [hunk 1]
```diff
@@ -3,174 +3,181 @@ import { getCurrentState } from '../store/actions/actions';
...
+                        if (layer.popup.body) {
+                            content += `<div><center>${Mustache.render(layer.popup.body, formattedData)}</center></div>`;
+                        }
+                    } else {
+                        content = Mustache.render(
+                            layer.popup.body,
+                            formattedData,
+                        );
+                    }
+                }
```
Comment (kahummer) [inline src/map/addPopUp.js]: @cKellyDesign I branches off from staging and I think prettier is the one formatting my code in this case

File: src/map/addPopUp.js [hunk 2]
```diff
@@ -3,174 +3,181 @@ import { getCurrentState } from '../store/actions/actions';
...
+                            (!layer.popup.join &&
+                                (row[layer.source.join[1]] === feature.properties[layer.source.join[0]]))) {
+                            const datum = {
+                                ...rowItem,
+                            };
+                            if (layer.popup && layer.popup.hideNulls) {
+                                Object.keys(datum).forEach((k) => {
+                                    if (Number.isNaN(Number(datum[k])) &&
+                                        (datum[k].toLowerCase() === 'n/a' ||
+                                            datum[k] === '')) {
```
Comment (cKellyDesign) [inline src/map/addPopUp.js]: It feels like we don't need to check if `datum[k]` is a number if we are also going to check if it's these strings?

Comment (cKellyDesign) [reply src/map/addPopUp.js]: Also, should we add to this conditional so it actually checks for `null`?

File: src/map/addPopUp.js [hunk 3]
```diff
@@ -3,174 +3,181 @@ import { getCurrentState } from '../store/actions/actions';
...
+                                    });
+                                    content =
+                                        `<div>` +
+                                        `<div><b>${row[layer.popup.header]}</b></div>` +
+                                        `<div><center>${Mustache.render(
+                                            layer.popup.body,
+                                            commaFormatting(layer, rowItem, true)
+                                        )}</center></div>` +
+                                        `</div>`;
+                                }
```
Comment (cKellyDesign) [inline src/map/addPopUp.js]: Why do we need two blocks here? Why do we not do check for `,` if we're hiding nulls?

File: src/map/addPopUp.js (hunk 2 above)
Comment (kahummer) [reply src/map/addPopUp.js]: @cKellyDesign I think n/a and ' ' are most common in data sets we handle. We surely can include null on the list. In terms of checking if it's not a number that's a good call we really don't need to check for that :fist_oncoming:

File: src/map/addPopUp.js (hunk 3 above)
Comment (kahummer) [reply src/map/addPopUp.js]: I will need to refactor this code

File: src/map/addPopUp.js [hunk 4]
```diff
@@ -2,175 +2,190 @@ import Mustache from 'mustache';
 import { getCurrentState } from '../store/actions/actions';
 import commaFormatting from './../utils/commaFormatting';
 
+/**
+ *  Adds popup to the map
+ * @param {''} mapId - Identified what map to show popup on
+ * @param {{}} mapboxGLMap mapbox map object allows us to interact with map methods from mapbox
+ * @param {() => {}} dispatch redux function that accepts an action
```
Comment (cKellyDesign) [inline src/map/addPopUp.js]: Should we use `string`, `object`, `function` for the param typings?

Comment (kahummer) [reply src/map/addPopUp.js]: you're right changing that

**Pull request merged**